mongo/integration/csot_cse_prose_test.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) MongoDB, Inc. 2022-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+//go:build cse
+// +build cse
+
+package integration
+
+import (
+	"context"
+	"testing"
+
+	"go.mongodb.org/mongo-driver/event"
+	"go.mongodb.org/mongo-driver/internal/testutil"
+	"go.mongodb.org/mongo-driver/internal/testutil/assert"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/integration/mtest"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+// CSOT prose tests that require the 'cse' Go build tag.
+func TestCSOTClientSideEncryptionProse(t *testing.T) {
+	verifyClientSideEncryptionVarsSet(t)
+	mt := mtest.New(t, mtest.NewOptions().MinServerVersion("4.2").CreateClient(false))
+	defer mt.Close()
+
+	mt.RunOpts("1. maxTimeMS is not set for commands sent to mongocryptd",
+		noClientOpts, func(mt *mtest.T) {
+			kmsProviders := map[string]map[string]interface{}{
+				"local": {
+					"key": localMasterKey,
+				},
+			}
+			mongocryptdSpawnArgs := map[string]interface{}{
+				"mongocryptdSpawnArgs": []string{"--port=23000"},
+			}
```

Comment:
This may resolve test failures on 6.0 and later. Tests on < 6.0 may not be failing by connecting to mongocryptd spawned on port 27020.

mongocryptd is a singleton process. If mongocryptd detects another process using the pidfilepath, the process will immediately terminate. Other tests spawn mongocryptd on port 27020 using the default pidfilepath.

```suggestion
			mongocryptdSpawnArgs := map[string]interface{}{
				// Pass a custom pidfilepath to ensure a new mongocryptd process is spawned.
				"mongocryptdSpawnArgs": []string{"--port=23000", "--pidfilepath=TestCSOTClientSideEncryptionProse_1.pid"},
				"mongocryptdUri":       "mongodb://localhost:23000",
				// Do not use the shared library to ensure mongocryptd is spawned.
				"__cryptSharedLibDisabledForTestOnly": true,
			}
```

Reply:
Sounds exactly right! Thanks 🧑‍🔧 .